Add FAQ for case sensitivity problem with MySQL (#6127)
optuna/optuna: docs/source/faq.rst

        for artifact_meta in get_all_artifact_meta(study):
            # Remove the artifacts uploaded to ``base_path``.
            artifact_store.remove(artifact_meta.artifact_id)

How can I resolve case sensitivity issues with MySQL?

> ------------------------------------------------------

Reviewer: ```suggestion
-----------------------------------------------------
```

By default, MySQL performs case-insensitive string comparisons.

> However, Optuna treats string parameters in a case-sensitive manner.
> This can lead to conflicts when parameter names differ only by case.

Reviewer: ```suggestion
However, Optuna treats string parameters in a case-sensitive manner, leading to conflicts in MySQL if parameter names differ only by case.
```

For example,

.. code-block:: python

    def objective(trial):
        a = trial.suggest_int("a", 0, 10)
        A = trial.suggest_int("A", 0, 10)
        return a + A

> In this case, "a" and "A" are treated as distinct by Optuna, but MySQL considers them the same due to its default collation settings.

Reviewer: ```suggestion
In this case, Optuna treats `a` and `A` distinctively, but MySQL does not due to its default collation settings.
```

> As a result, an error may occur when trying to register these parameters in the database.

Reviewer: ```suggestion
As a result, only one of the parameters will be registered in MySQL.
```

> To address this issue, there are some workarounds.

Reviewer: ```suggestion
Some workarounds to be considered are as follows:
```

1. Use a different storage backend.
    Please consider using PostgreSQL or SQLite, which supports case-sensitive handling.
2. Rename the parameters to avoid case conflicts.

>     For example, use "a" and "b" instead of "a" and "A".

Reviewer: ```suggestion
    For example, use `a` and `b` instead of `a` and `A`.
```

3. Change MySQL’s collation settings to be case-sensitive.
    You can configure case sensitivity at the database, table, or column level.

>     For more details, refer to `the MySQL documentation <https://dev.mysql.com/doc/refman/9.3/en/charset-syntax.html>`__.

Reviewer: ```suggestion
    We defer to `the MySQL documentation <https://dev.mysql.com/doc/refman/9.3/en/charset-syntax.html>`__ for more details.
```
